backend/src/main/java/meetup/connect/auth/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,71 @@
+package meetup.connect.auth;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+
+
+import java.io.IOException;
+
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+  private static final String HTTP_AUTHORIZATION_HEADER = "Authorization";
+  private static final String BEARER_TOKEN_PREFIX = "Bearer ";
+  private static final int BEARER_TOKEN_PREFIX_LENGTH = 7;
+
+  private final JwtService jwtService;
+  private final UserDetailsService userDetailsService;
+
+  public JwtAuthenticationFilter(JwtService jwtService, UserDetailsService userDetailsService) {
+    this.jwtService = jwtService;
+    this.userDetailsService = userDetailsService;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      @NonNull HttpServletRequest request,
+      @NonNull HttpServletResponse response,
+      @NonNull FilterChain filterChain)
+      throws ServletException, IOException {
+    final String authHeader = request.getHeader(HTTP_AUTHORIZATION_HEADER);
+    final String jwtToken;
+    final String userEmail;
+
+    // Check JWT Token
+    if (authHeader == null || !authHeader.startsWith(BEARER_TOKEN_PREFIX)) {
+      // execute another filter if there isn't jwt Token
+      filterChain.doFilter(request, response);
+      return;
+    }
+
+    jwtToken =
+        authHeader.substring(
+            BEARER_TOKEN_PREFIX_LENGTH); // Extract token that starts with Bearer: token
+    System.out.println(jwtToken);
```

Comment:
issue: please, don't use `System.out.println` in the production code

Reply:
use the `@Slf4j` annotation from lombok, and `logger.info()`

Reply:
note: you can read further about the logging levels: https://sematext.com/blog/logging-levels/